p2p/protocol/identify/id.go

```diff
@@ -252,8 +252,10 @@ func (ids *IDService) consumeMessage(mes *pb.Identify, c inet.Conn) {
 	ids.addrMu.Lock()
 	switch ids.Host.Network().Connectedness(p) {
 	case inet.Connected:
+		ids.Host.Peerstore().UpdateAddrs(p, pstore.ConnectedAddrTTL, 0)
```

Comment:
I wasn't actually suggesting that we _use_ `UpdateAddrs` (I was also thinking about fixing `SetAddrs`) but I actually like this more.

Maybe we should downgrade these to `TempAddrTTL` instead of 0? Honestly, I'm fine either way and `TempAddrTTL` is long enough that that might cause problems. My only concern is that we'll have a short period where we'll have _no_ good addresses.

We could also use a constant 10s timeout (for _really_ temporary addresses).

Reply:
It's a valid concern I think, so I'll update to use a 10s ttl to cover the gap.

Reply:
I called it `transientTTL` and it's 10s long; maybe we should have this constant in the peerstore.

Reply:
Well, that's what `TempAddrTTL` was _supposed_ to be...